src/components/Map/utils.ts

```diff
@@ -111,3 +115,37 @@ export const getMapDrawingLayer = (source: MapSourceRenderProperties, sourceId:
       : null,
   };
 };
+
+/**
+ * Cut source polygons into smaller polygons
+ * @param source
+ *  The source polygons that need to be split into smaller polygons
+ * @param cutWith
+ *  The polygon to use as the overlap check in order to cut polygons
+ * @returns
+ *  List of polygons that were cut as a result of overlaps with cutWith
+ *  Null if there was no overlap
+ */
+export const cutPolygons = (source: GeometryFeature[], cutWith: Geometry): GeometryFeature[] | null => {
```

Comment:
Can you add a test for this?

Reply:
This code is not done yet, it is still evolving. I created a [JIRA](https://terraformation.atlassian.net/browse/SW-4715) and will add a test when it is ready. That ok?